### PR TITLE
A few more strong-mode fixes

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: test/pub_uploader_test.dart [hunk 1]
```diff
@@ -31,7 +31,7 @@ See http://dartlang.org/tools/pub/cmd/pub-uploader.html for detailed documentati
 ScheduledProcess startPubUploader(ScheduledServer server, List<String> args) {
   var tokenEndpoint = server.url.then((url) =>
       url.resolve('/token').toString());
-  args = ['uploader', '--server', tokenEndpoint]..addAll(args);
+  args = <Object>['uploader', '--server', tokenEndpoint]..addAll(args);
```
Comment (nex3) [inline test/pub_uploader_test.dart]: Why does this need to be annotated as `Object`? Isn't that the LUB for `Future<String>` and `String`?

File: lib/src/barback.dart [hunk 2]
```diff
@@ -34,7 +34,7 @@ import 'package:pub_semver/pub_semver.dart';
 /// constraint and barback's version.
 ///
 /// [compat]: https://gist.github.com/nex3/10942218
-final pubConstraints = {
+final pubConstraints = <String, VersionRange>{
```
Comment (nex3) [inline lib/src/barback.dart]: There's no guarantee that `new VersionConstraint.parse()` will emit a `VersionRange`, so this annotation will probably end up generating an implicit cast warning at some point. If we rely on these being `VersionRange`s, you should explicitly cast them with `as`.

**Comment by nex3** [comment]
@kevmoo Are you planning on iterating on this?

File: lib/src/barback.dart (hunk 2 above)
Comment (kevmoo) [inline lib/src/barback.dart]: True. But in every case where you call in in this final, it is a `VersionRange`. Moreover, in every usage of `pubConstraints` you assume the type is `VersionRange`.

Seems better to fail here than elsewhere, no?

**Comment by kevmoo** [comment]
I replied to your requests. Happy to discuss.

File: test/pub_uploader_test.dart (hunk 1 above)
Comment (nex3) [inline test/pub_uploader_test.dart]: Go on...

**Comment by kevmoo** [comment]
@nex3 PTAL?

**Comment by nex3** [comment]
There are outstanding code review comments...

File: test/pub_uploader_test.dart (hunk 1 above)
Comment (kevmoo) [inline test/pub_uploader_test.dart]: `String` is not the LUB for `Future<String>` and `String`. Hence, `Object`.

Comment (nex3) [inline test/pub_uploader_test.dart]: I'm asking why `Object` isn't the LUB.

File: lib/src/barback.dart (hunk 2 above)
Comment (kevmoo) [inline lib/src/barback.dart]: PTAL

File: test/pub_uploader_test.dart (hunk 1 above)
Comment (kevmoo) [inline test/pub_uploader_test.dart]: Let's talk in person – I'm not sure what you're asking...

**Pull request merged**